src/QueryRenderer.ts

```diff
@@ -496,19 +494,16 @@ class QueryRenderChild extends MarkdownRenderChild {
             newTasks,
         });
 
-        const postponedDateString = postponedDate?.format('DD MMM YYYY');
-        this.onPostponeSuccessCallback(button, dateTypeToUpdate, postponedDateString);
+        this.onPostponeSuccessCallback(button, dateTypeToUpdate, postponedDate);
     }
 
-    private onPostponeSuccessCallback(
-        button: HTMLButtonElement,
-        updatedDateType: HappensDate,
-        postponedDateString: string,
-    ) {
+    private onPostponeSuccessCallback(button: HTMLButtonElement, updatedDateType: HappensDate, postponedDate: Moment) {
         // Disable the button to prevent update error due to the task not being reloaded yet.
         button.disabled = true;
         button.setAttr('title', 'You can perform this action again after reloading the file.');
-        new Notice(`Task's ${updatedDateType} postponed untill ${postponedDateString}`, 5000);
+
+        const postponedDateString = postponedDate?.format('DD MMM YYYY');
+        new Notice(`Task's ${updatedDateType} postponed until ${postponedDateString}`, 5000);
```

Comment:
The string displayed in this message may grow to contain an increasing amount of logic.

Personally, I would have extracted the initial string to a function and tested it first, before making the change.

But given where this code is now, I would extract the new behaviour to a helper function, and test it, so at least we have it protected, and easier to modify-and-test in future iterations.

Reply:
I'm not really sure what shall be extracted here. As for me I would inline the `onPostponeSuccessCallback` first and then look for something to extract. I think this is ok for now as it is we can improve further during our session.

Reply:
Ok, clear, thanks.

Reply:
Perfect. Thank you. Definitely good enough... I'm not sure any further improvement is needed...